### PR TITLE
Fixes lib machines spamming refreshes when printing

### LINKED ISSUE #7704: Printing a book via SS13BN causes the book list to reload itself 9 times before printing the book
it's bad enough even without the game-breaking lag that one reload causes.

### LINKED ISSUE #8664: Printing a book via SS13BN causes several print attempts
if it's a valid number, it'll try and print it three times. the first is successful, the second and third are not because of that anti-spam thing.
age-old bug that probably should have been fixed some time ago.

### PR DESCRIPTION
**Pull request by MrsTonedOne**
fixes #8664
fixes #7704

For shame @Chocobro making a dupe of your own issue.

### CODE REVIEW

File: code/modules/library/lib_machines.dm [hunk 1]
```diff
@@ -433,12 +437,6 @@ var/global/list/datum/cachedbook/cachedbooks // List of our cached book datums
 				B.icon_state = "book[rand(1,7)]"
 				src.visible_message("[src]'s printer hums as it produces a completely bound book. How did it do that?")
 				break
-	if(href_list["orderbyid"])
-		var/orderid = input("Enter your order:") as num|null
-		if(orderid)
-			if(isnum(orderid))
-				var/nhref = "src=\ref[src];targetid=[orderid]"
-				spawn() src.Topic(nhref, params2list(nhref), src)
```
Comment (RemieRichards) [inline code/modules/library/lib_machines.dm]: ahahahaha what.
is this pre ```updateUsrDialog()``` code?

Comment (MrStonedOne) [inline code/modules/library/lib_machines.dm]: It's pre-everything.

This code is new enough that goon coders never worked on it, but old enough that babby tgcoders worked on it before the knew what maintainability is.

Comment (RemieRichards) [inline code/modules/library/lib_machines.dm]: glorious.

**Comment by bgobandit** [comment]
:+1: x100

File: code/modules/library/lib_machines.dm [hunk 2]
```diff
@@ -406,7 +406,11 @@ var/global/list/datum/cachedbook/cachedbooks // List of our cached book datums
 						else
 							log_game("[usr.name]/[usr.key] has uploaded the book titled [scanner.cache.name], [length(scanner.cache.dat)] signs")
 							alert("Upload Complete. Uploaded title will be unavailable for printing for a short period")
-
+	if(href_list["orderbyid"])
+		var/orderid = input("Enter your order:") as num|null
+		if(orderid)
+			if(isnum(orderid))
+				href_list["targetid"] = orderid
 	if(href_list["targetid"])
```
Comment (tkdrg) [inline code/modules/library/lib_machines.dm]: Have you tested this? I'm not 100% sure of the semantics of this in DM... Does it work like C's `switch()`es and fall down to the next one unless returned?

Comment (RemieRichards) [inline code/modules/library/lib_machines.dm]: this isn't a switch.
so it's just normal ifs after other ifs.

File: code/modules/library/lib_machines.dm (hunk 1 above)
Comment (Iamgoofball) [inline code/modules/library/lib_machines.dm]: This needs archived or some shit, this is art

**Pull request merged**